### DISCUSSION TITLE
Highligh specific cells based on condition

### DISCUSSION THREAD

**Discussion by wdv4758h** [nushell/nushell#8989] [Show and tell]
A small sharing.

I'm thinking of hightlight some cells based on condition. Still showing the whole table, but mark some cells to make it more obvious (e.g. data that need more attention).

Not sure if there is better way. But I got a quick result with `update cells` plus `ansi`.

![image](https://user-images.githubusercontent.com/2716047/234243259-3779cc56-bddc-4f16-abe2-7daab7372b27.png)

```nu
[[A, B]; [a b] [c d]]
| update cells {|i| if ($i == b) { $"(ansi -e '2;30;43m')($i)(ansi reset)" } else { $i }}
```

However, this will mess with the data. And the value will be string. If the original type was int, the type will become any (since it needs to accept both string and int). Will need to cache the original data if there is following processing.

**Comment by fdncred** [comment]
Table items are currently highlighted by datatype.
![image](https://user-images.githubusercontent.com/343840/234269556-f967a3b5-b6bc-4e64-815e-2965e1dd7709.png)
You can see that when I change the second row, the colors change. You can set the datatype colors in a theme.

However, I do not believe there is any other way to change the color of an individual cell within datatype other than with ansi escapes.
